### PR TITLE
v6.0.0 - Add experimental pomerium-operator support, deprecate legacy TLS Secret support

### PR DESCRIPTION
**Pull request by travisgroth**

### CODE REVIEW

File: charts/pomerium/README.md [hunk 1]
```diff
@@ -57,6 +58,16 @@ helm delete --purge my-release
 
 The command removes nearly all the Kubernetes components associated with the chart and deletes the release.
 
+## Pomerium Operator (EXPERIMENTAL)
```
Comment (desimone) [inline charts/pomerium/README.md]: 🔥

File: charts/pomerium/templates/authorize-deployment.yaml [hunk 2]
```diff
@@ -54,10 +56,9 @@ spec:
         image: {{ .Values.image.repository }}:{{ .Values.image.tag }}
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         args:
-{{- if or .Values.config.existingConfig .Values.config.policy }}
+        {{- if or (or .Values.config.existingConfig .Values.config.policy) .Values.operator.enabled }}	
           - --config=/etc/pomerium/config.yaml
-{{- end }}
-{{- range $key, $value := .Values.extraArgs }}
+        {{- end }}{{- range $key, $value := .Values.extraArgs }}
```
Comment (desimone) [inline charts/pomerium/templates/authorize-deployment.yaml]: nit:  I think the for loop pops out a bit more on a seperate line

File: charts/pomerium/templates/configmap.yaml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {{- if not .Values.config.existingConfig }}
+{{- if .Values.operator.enabled }}
```
Comment (desimone) [inline charts/pomerium/templates/configmap.yaml]: This looks off. We'd lose these annotations in non operator mode right?

Comment (travisgroth) [reply charts/pomerium/templates/configmap.yaml]: The logic here is a little odd.  In operator mode we have two config maps - one empty, and one populated from the static values of the chart.  When running without the operator, we only need the main config map with static values.

To save a big repeated block in the conditional, I am changing the name of the ConfigMap with the static values depending on whether we're in operator mode.  Similarly, I just flip the empty ConfigMap off in non-operator mode.

Hopefully that tracks what's in the yaml...unless I'm missing something else.

File: charts/pomerium/templates/authorize-deployment.yaml (hunk 2 above)
Comment (travisgroth) [reply charts/pomerium/templates/authorize-deployment.yaml]: Good catch.

File: charts/pomerium/README.md [hunk 4]
```diff
@@ -157,9 +168,24 @@ A full listing of Pomerium's configuration variables can be found on the [config
...
+| `operator.config.debug`               | Enable Pomerium Operator debug logging                                                                                                                                                                                                                                                             | `false`                                                                               |
+| `operator.deployment.annotations`     | Annotations for the operator deployment.                                                                                                                                                                                                                                                           | `{}`                                                                                  |
 
 ## Changelog
 
+### 6.0.0
+
+- Integrate pomerium operator
+- Remove legacy TLS config support.  See [v3.0.0 Upgrade Notes](#300-1) to migrate
```
Comment (desimone) [inline charts/pomerium/README.md]: 👍

File: charts/pomerium/templates/configmap.yaml (hunk 3 above)
Comment (desimone) [reply charts/pomerium/templates/configmap.yaml]: Makes  complete sense. 👍

**Pull request merged**